### PR TITLE
Removing LTO from debugging builds

### PR DESCRIPTION
**Pull request by kneth**
Using a locally build core, it is possible to single step into C++. You will have to add the following two lines to `realm.properties`:

    debugBuild=true
    stripSymbols=false

@cmelchior @emanuelez @beeender

### CODE REVIEW

File: realm/realm-jni/build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ import java.security.MessageDigest
 
 ext.coreVersion = '0.95.7'
 // empty or comment out this to disable hash checking
-ext.coreSha256Hash = '11c82f5da9b43b62fbf55fbd3aa61dc6956bc3483bbaa5f69262018144c8cc3c'
+//ext.coreSha256Hash = '11c82f5da9b43b62fbf55fbd3aa61dc6956bc3483bbaa5f69262018144c8cc3c'
```
Comment (cmelchior) [inline realm/realm-jni/build.gradle]: Is this a mistake?

File: realm/realm-jni/build.gradle [hunk 2]
```diff
@@ -15,7 +15,11 @@ ext.stripSymbols = project.hasProperty('stripSymbols') ? project.getProperty('st
 // core source code.
 ext.coreSourcePath = project.hasProperty('coreSourcePath') ? project.getProperty('coreSourcePath') : null
 
-def commonCflags = [ '-Os', '-std=c++11', '-ffunction-sections', '-fdata-sections', '-flto' ]
+def commonCflags = [ '-Os', '-std=c++11' ]
+// LTO and debugging don't play well together
+if (!ext.debugBuild) {
+    commonCflags += [ '-ffunction-sections', '-fdata-sections', '-flto' ]
```
Comment (emanuelez) [inline realm/realm-jni/build.gradle]: should visibility hidden be here?

**Comment by kneth** [comment]
@emanuelez @cmelchior Thanks for the comments. It is work-in-progress, and I merely wished to show you how to do native debugging.

**Comment by kneth** [comment]
Since migrating to the new experimental plugin seems to be too hard (other plugins won't work together with it), this is what we can do to get native debugging work.

@realm/java

**Comment by dalinaum** [comment]
:+1:

**Comment by beeender** [comment]
:+1:

**Comment by emanuelez** [comment]
Can we make it so that if a build a debug one then symbols are kept by default?

**Comment by kneth** [comment]
@emanuelez You mean, that `buildDebug = true` implies `stripSymbols = false`?

**Comment by emanuelez** [comment]
Yes

**Comment by beeender** [comment]
@emanuelez @kneth It has already been like that.

```
 if (ext.debugBuild) {
     // Debug build should never strip symbols
     ext.stripSymbols = false
 }
```

**Pull request merged**